core-bundle/src/Resources/contao/controllers/FrontendIndex.php

```diff
@@ -181,9 +181,9 @@ public function renderPage($pageModel)
 		if ($objPage->alias != '')
 		{
 			$language = \Config::get('addLanguageToUrl') ? '[a-z]{2}(-[A-Z]{2})?/' : '';
-			$suffix = \Config::get('urlSuffix') ? preg_quote(\Config::get('urlSuffix'), '#') : '$';
+			$suffix = \Config::get('urlSuffix') ? preg_quote(\Config::get('urlSuffix'), '#') : '';
 
-			if (preg_match('#^' . $language . $objPage->id . '(' . $suffix . '|/)#', \Environment::get('relativeRequest')))
+			if (preg_match('#^' . $language . $objPage->id . '(' . $suffix . '($|\?)|/)#', \Environment::get('relativeRequest')))
```

Comment:
Should we not add this in line 184 instead?

```php
$suffix = \Config::get('urlSuffix') ? preg_quote(\Config::get('urlSuffix'), '#') : '($|\?)';
```

Or, if it does not matter whether `$suffix` is empty or not, should we not simplify line 184?

```php
$suffix = preg_quote(\Config::get('urlSuffix'), '#');
```

Reply:
> Should we not add the ($|\?) in line 184 instead?

no

> Or, if it does not matter whether $suffix is empty or not, should we not simplify line 184?

why would you `preg_quote` an empty string?

Reply:
The check is not necessary, because `preg_quote()` already checks and returns early if the string is empty:

https://github.com/php/php-src/blob/aa79a22d32f31dbc5343f8191e925aa07447c3ec/ext/pcre/php_pcre.c#L2734-L2737